File: java/src/main/java/codingdojo/BasicItem.java
package codingdojo;

public class BasicItem implements Item {

    private String name;
    private int baseDamage;
    private float damageModifier;

    public BasicItem(String name, int baseDamage, float damageModifier) {
        this.name = name;
        this.baseDamage = baseDamage;
        this.damageModifier = damageModifier;
    }

    @Override
    public int getBaseDamage() {
        return baseDamage;
    }

    @Override
    public float getDamageModifier() {
        return damageModifier;
    }
}

File: cpp/model/BasicItem.cpp
#include "BasicItem.h"

BasicItem::BasicItem(const std::string &name, int baseDamage, float damageModifier) : name(name), baseDamage(baseDamage), damageModifier(damageModifier) {

}

const float BasicItem::getDamageModifier() {
    return damageModifier;
}

const int BasicItem::getBaseDamage() {
    return baseDamage;
}
